src/renderer/hooks/usePaneNodeSearchMenu.tsx

```diff
@@ -0,0 +1,417 @@
+                                                    accentColor="gray.500"
+                                                    icon={node.icon}
+                                                />
+                                                <Text
+                                                    h="full"
+                                                    verticalAlign="middle"
+                                                >
+                                                    {node.name}
+                                                </Text>
+                                                (
+                                                {isFavorite && (
+                                                    <>
+                                                        <Spacer />
+                                                        <StarIcon
+                                                            aria-label="Favorites"
+                                                            boxSize={2.5}
+                                                            color="gray.500"
+                                                            overflow="hidden"
+                                                            stroke="gray.500"
+                                                            verticalAlign="middle"
+                                                        />
+                                                    </>
+                                                )}
+                                                )
+                                            </HStack>
+                                        );
+                                    })}
+                                </Box>
+                            );
+                        })
+                    ) : (
+                        <Center
+                            opacity="50%"
+                            w="full"
+                        >
+                            No compatible nodes found.
+                        </Center>
+                    )}
+                </Box>
+            </MenuList>
+        ),
+        [
+            connectingFrom,
+            connectingFromType,
+            byCategories,
+            onPaneContextMenuNodeClick,
+            searchQuery,
+            schemata.schemata,
+            matchingNodes,
+            mousePosition,
+        ]
+    );
+
+    useEffect(() => {
+        if (connectingFrom) {
+            const { nodeId, inOutId } = parseHandle(connectingFrom.handleId!);
+            const node: Node<NodeData> | undefined = getNode(nodeId);
+            if (node) {
+                if (connectingFrom.handleType === 'source') {
+                    const sourceType = functionDefinitions
+                        .get(node.data.schemaId)
+                        ?.outputDefaults.get(inOutId);
+                    setConnectingFromType(sourceType!);
```

Comment:
`setConnectingFromType` accepts `null`, so why not use that?

```suggestion
                    setConnectingFromType(sourceType ?? null);
```